lib/adagrams.rb

```diff
@@ -0,0 +1,136 @@
+def draw_letters
+  letter_quantities = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1
+  }
+  
+  # places the appropriate number of each letter in an array
+  letters_array = []
+  letter_quantities.each do |key, value|
+    value.times do
+      letters_array << key.to_s
+    end
+  end
+
+  return letters_array.shuffle.first(10)
+end
+
+def uses_available_letters?(input, letters_in_hand)
+
+  # array of booleans to store results of each iteration through the input string
+  include_letter_array = []
+
+  # duplicate of the 10 drawn letters, in order to not alter the original array
+  cloned_letter_in_hand = letters_in_hand.clone
+
+  input.each_char do | letter |
+    if cloned_letter_in_hand.include?(letter)
+      # deletes letter at index (not all instances of the letter)
+      cloned_letter_in_hand.slice!(cloned_letter_in_hand.index(letter))
+      include_letter_array << true
+    else
+      include_letter_array << false
+    end
+  end
+  # evaluates boolean array: if all values are true, returns true. Otherwise returns false
+  return include_letter_array.all?(true)
+end
+
+
+def score_word(word)
+  sum = 0
+
+  word.upcase.each_char do |char|
+
+    case char
```

Comment:
While this `case` statement works, it means that the information about which letter has which score is locked into this piece of code, and can't easily be used elsewhere. For example, if you wanted to display the value of each letter in a hand, you would need to repeat this work.

An alternative approach would be to store the letter scores in a hash, something like this:

```ruby
LETTER_SCORES = {
  "A" => 1
  "B" => 3,
  "C" => 3,
  "D" => 2,
  # ...
}
```

Then to get the score for a letter, you can say `LETTER_SCORES[letter]`.